src/lib/datepicker/month-view.ts

```diff
@@ -46,12 +59,16 @@ export class MatMonthView<D> implements AfterContentInit {
   @Input()
   get activeDate(): D { return this._activeDate; }
   set activeDate(value: D) {
-    let oldActiveDate = this._activeDate;
-    this._activeDate =
+    const oldActiveDate = this._activeDate;
+    const validDate =
         this._getValidDateOrNull(this._dateAdapter.deserialize(value)) || this._dateAdapter.today();
+    this._activeDate = this._dateAdapter.clampDate(validDate, this.minDate, this.maxDate);
     if (!this._hasSameMonthAndYear(oldActiveDate, this._activeDate)) {
       this._init();
     }
+    if (this._dateAdapter.compareDate(oldActiveDate, this._activeDate)) {
+      this.activeDateChange.emit(this._activeDate);
```

Comment:
are we sure we want to emit here? it seems strange to emit in response to the `@Input()` changing. I would have expected it to only emit if the active date changed due to user interaction (click, keyboard, etc)

Reply:
It's here for convenience (just because there was a single line above saving the oldActiveDate), but you're right. This is far from being the best place to put it. I'll move it.